src/hooks/useTables.ts

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect, useRef } from 'react';
+import { Datasource } from 'data/CHDatasource';
+
+export default (datasource: Datasource, database: string): string[] => {
+  const [tables, setTables] = useState<string[]>([]); 
+  
+  useEffect(() => {
+    if (!datasource || !database) {
+      return;
+    }
+
+    let ignore = false;
```

Comment:
In what cases `ignore` should be true?

Reply:
Added this to prevent race conditions when setting the state with the async table results. Must've shifted a line around, the cleanup effect incorrectly had `ignore = false` when it should've been `ignore = true`